### PR TITLE
chore(visibility): Manually instrument tagstore cache

### PR DESCRIPTION
**Pull request by gggritso**
We have Django cache instrumentation turned off, but I want to track the cache hit rate for tagstore in Sentry itself if I can. Firstly to dogfood, and secondly to help with some caching improvements I'm thinking of making.

**e.g.,**

Screenshot 2024-11-26 at 2 17 56 PM

### CODE REVIEW

File: src/sentry/tagstore/snuba/backend.py [hunk 1]
```diff
@@ -309,11 +310,19 @@ def __get_tag_keys_for_projects(
             # Cause there's rounding to create this cache suffix, we want to update the query end so results match
             end = snuba.quantize_time(end, key_hash)
             cache_key += f":{duration}@{end.isoformat()}"
-            result = cache.get(cache_key, None)
-            if result is not None:
-                metrics.incr("testing.tagstore.cache_tag_key.hit")
-            else:
-                metrics.incr("testing.tagstore.cache_tag_key.miss")
+
+            with sentry_sdk.start_span(op="cache.get", name="tagstore.cache") as span:
```
Comment (DominikB2014) [inline src/sentry/tagstore/snuba/backend.py:314]: Just a note, if we instrument more, we'll probably have to standardize the span description `<cache namespace>.cache`. Not sure if namespace is the right word lol, but do you get what i mean?

**Review by DominikB2014 — APPROVED** [review]
lgtm, just a small note!

File: src/sentry/tagstore/snuba/backend.py [hunk 2]
```diff
@@ -309,11 +310,19 @@ def __get_tag_keys_for_projects(
...
+                result = cache.get(cache_key, None)
+
+                span.set_data("cache.key", cache_key.split(":", 2))
```
[the remark this answers is not in the document]
Comment (gggritso) [reply src/sentry/tagstore/snuba/backend.py]: Gotcha! I misunderstood the docs, fixed!

File: src/sentry/tagstore/snuba/backend.py (hunk 1 above)
Comment (gggritso) [reply src/sentry/tagstore/snuba/backend.py]: I made the namespace more specific, should be better now!

**Pull request merged**